Broken Pipe error logs in Ember Server when using `withTLS`

similar to #7014 but for `withTLS` instead of `withHttp2`

```
scala 3.3.1
org.typelevel:cats-effect_3:3.5.2
co.fs2:fs2-core_3:3.9.3
org.http4s:http4s-core_3:0.23.25
```

I get error logs when killing curl requests against ember with TLS enabled:

``` scala
EmberServerBuilder
        .default[IO]
        .withHost(ipv4"0.0.0.0")
        .withPort(port"8443")
        .withTLS(tlsContext)
        .withHttpApp(finalHttpApp)
        .build
```
Error log:
```
root 2024-01-10 09:33:34,714 [ERROR]             [io-compute-10] o.h.e.server.EmberServerBuilderCompanionPlatform - Request handler failed with exception
root java.io.IOException: Broken pipe
root    at java.base/sun.nio.ch.SocketDispatcher.write0(Native Method)
root    at java.base/sun.nio.ch.SocketDispatcher.write(SocketDispatcher.java:62)
root    at java.base/sun.nio.ch.IOUtil.writeFromNativeBuffer(IOUtil.java:137)
root    at java.base/sun.nio.ch.IOUtil.write(IOUtil.java:102)
root    at java.base/sun.nio.ch.IOUtil.write(IOUtil.java:65)
root    at java.base/sun.nio.ch.UnixAsynchronousSocketChannelImpl.finishWrite(UnixAsynchronousSocketChannelImpl.java:602)
root    at java.base/sun.nio.ch.UnixAsynchronousSocketChannelImpl.finish(UnixAsynchronousSocketChannelImpl.java:199)
root    at java.base/sun.nio.ch.UnixAsynchronousSocketChannelImpl.onEvent(UnixAsynchronousSocketChannelImpl.java:217)
root    at java.base/sun.nio.ch.EPollPort$EventHandlerTask.run(EPollPort.java:305)
root    at java.base/sun.nio.ch.AsynchronousChannelGroupImpl$1.run(AsynchronousChannelGroupImpl.java:113)
root    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
root    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
root    at java.base/java.lang.Thread.run(Thread.java:1583)
root    at delay @ fs2.io.net.SocketCompanionPlatform$AsyncSocket.go$2$$anonfun$1(SocketPlatform.scala:131)
root    at async @ fs2.io.net.SocketCompanionPlatform$AsyncSocket.go$2(SocketPlatform.scala:131)
root    at flatMap @ fs2.io.net.SocketCompanionPlatform$AsyncSocket.go$2(SocketPlatform.scala:135)
root    at delay @ fs2.io.net.SocketCompanionPlatform$AsyncSocket.write(SocketPlatform.scala:138)
root    at flatMap @ fs2.io.net.SocketCompanionPlatform$AsyncSocket.write(SocketPlatform.scala:138)
root    at apply @ org.http4s.ember.server.EmberServerBuilder.build$$anonfun$2(EmberServerBuilder.scala:216)
```

Broken pipes are errors, it is fine to log them but error loglevel is a little much in my opinion. I believe https://github.com/http4s/http4s/pull/7016 fixed something similar but it seems not to work for TLS.

--- a/ember-server/shared/src/main/scala/org/http4s/ember/server/internal/ServerHelpers.scala
+++ b/ember-server/shared/src/main/scala/org/http4s/ember/server/internal/ServerHelpers.scala
@@ -194,7 +194,9 @@ private[server] object ServerHelpers extends ServerHelpersPlatform {
         val handler: Stream[F, Nothing] = shutdown.trackConnection >>
           Stream
             .resource(upgradeSocket(connect, tlsInfoOpt, logger, enableHttp2))
-            .handleErrorWith(err => Stream.exec(logger.warn(err)("Failed to upgrade socket to TLS")))
+            .handleErrorWith(err =>
+              Stream.exec(logger.warn(err)("Failed to upgrade socket to TLS"))
+            )
             .flatMap {
               case (socket, Some("h2")) =>
                 // ALPN H2 Strategy

--- a/ember-server/shared/src/main/scala/org/http4s/ember/server/internal/ServerHelpers.scala
+++ b/ember-server/shared/src/main/scala/org/http4s/ember/server/internal/ServerHelpers.scala
@@ -194,6 +194,8 @@ private[server] object ServerHelpers extends ServerHelpersPlatform {
         val handler: Stream[F, Nothing] = shutdown.trackConnection >>
           Stream
             .resource(upgradeSocket(connect, tlsInfoOpt, logger, enableHttp2))
+            .onError(err => Stream.eval(logger.warn(err)("Failed to upgrade socket to TLS")))
+            .mask
             .flatMap {
               case (socket, Some("h2")) =>
                 // ALPN H2 Strategy

--- a/ember-server/shared/src/main/scala/org/http4s/ember/server/internal/ServerHelpers.scala
+++ b/ember-server/shared/src/main/scala/org/http4s/ember/server/internal/ServerHelpers.scala
@@ -194,8 +194,7 @@ private[server] object ServerHelpers extends ServerHelpersPlatform {
         val handler: Stream[F, Nothing] = shutdown.trackConnection >>
           Stream
             .resource(upgradeSocket(connect, tlsInfoOpt, logger, enableHttp2))
-            .onError(err => Stream.eval(logger.warn(err)("Failed to upgrade socket to TLS")))
-            .mask
+            .handleErrorWith(err => Stream.exec(logger.warn(err)("Failed to upgrade socket to TLS")))
             .flatMap {
               case (socket, Some("h2")) =>
                 // ALPN H2 Strategy

